market-generator/src/main/java/com/market/banica/generator/util/SnapshotPersistence.java

```diff
@@ -12,61 +12,111 @@
+        persistMarketSnapshot(marketSnapshot);
+
+    }
+
+    public void persistMarketSnapshot(Queue<MarketTick> marketSnapshot) throws IOException {
+
+        Output output = new Output(new FileOutputStream(ApplicationDirectoryUtil.getConfigFile(snapshotFileName)));
+        kryoHandle.writeClassAndObject(output, marketSnapshot);
         output.close();
-        LOGGER.info("Persisting snapshot database!");
+        LOGGER.debug("Persisting market snapshot!");
+
+    }
+
+    @SuppressWarnings({"unchecked"})
+    public Map<String, Set<MarketTick>> loadMarketState() throws IOException {
+
+        Map<String, Set<MarketTick>> loadedMarketStateTicks = new ConcurrentHashMap<>();
+
+        if (!ApplicationDirectoryUtil.doesFileExist(stateFileName)) {
+
+            LOGGER.info("Creating \"{}\" file!", stateFileName);
+            ApplicationDirectoryUtil.getConfigFile(stateFileName);
+
+        } else if (ApplicationDirectoryUtil.getConfigFile(stateFileName).length() == 0) {
+
+            LOGGER.info("File \"{}\" is empty, no ticks were loaded!", stateFileName);
+
+        } else {
+
+            Input input = new Input(new FileInputStream(ApplicationDirectoryUtil.getConfigFile(stateFileName)));
+            loadedMarketStateTicks = (Map<String, Set<MarketTick>>) kryoHandle.readClassAndObject(input);
+            input.close();
+            LOGGER.info("Loaded market state ticks!");
+
+        }
+        return loadedMarketStateTicks;
 
     }
 
     @SuppressWarnings({"unchecked"})
-    public Map<String, Set<MarketTick>> loadPersistedSnapshot() throws IOException {
+    public Queue<MarketTick> loadMarketSnapshot() throws IOException {
 
-        Map<String, Set<MarketTick>> loadedMarketTicks = new ConcurrentHashMap<>();
+        Queue<MarketTick> loadedSnapshotTicks = new LinkedBlockingQueue<>();
 
-        if (!ApplicationDirectoryUtil.doesFileExist(databaseFileName)) {
+        if (!ApplicationDirectoryUtil.doesFileExist(snapshotFileName)) {
 
-            LOGGER.info("Creating \"{}\" file!", databaseFileName);
-            ApplicationDirectoryUtil.getConfigFile(databaseFileName);
+            LOGGER.info("Creating \"{}\" file!", snapshotFileName);
+            ApplicationDirectoryUtil.getConfigFile(snapshotFileName);
 
-        } else if (ApplicationDirectoryUtil.getConfigFile(databaseFileName).length() == 0) {
+        } else if (ApplicationDirectoryUtil.getConfigFile(snapshotFileName).length() == 0) {
 
-            LOGGER.info("File \"{}\" is empty, no ticks were loaded!", databaseFileName);
+            LOGGER.info("File \"{}\" is empty, no ticks were loaded!", snapshotFileName);
 
         } else {
 
-            Input input = new Input(new FileInputStream(ApplicationDirectoryUtil.getConfigFile(databaseFileName)));
-            loadedMarketTicks = (Map<String, Set<MarketTick>>) kryoHandle.readClassAndObject(input);
+            Input input = new Input(new FileInputStream(ApplicationDirectoryUtil.getConfigFile(snapshotFileName)));
```

Comment:
is it sure that the ticks will be loaded in the correct order

Reply:
I believe so, due to the structure being a queue, kryo serializes the connections between the nodes so when polling it will always poll the element that was added first (FIFO - as the typical queue is) and keep the queue type structure in the dat file.